activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -128,7 +128,14 @@ def foreign_key_options
       def columns
         result = [["#{name}_id", type, options]]
         if polymorphic
-          result.unshift(["#{name}_type", :string, polymorphic_options])
+          options = polymorphic_options
+          if index
+            indexable_string_limit = Base.connection.max_indexable_string_limit
```

Comment:
Or we just use smaller limit like `100` any time. Because `"#{name}_type"` stores class name and class name isn't long.